### PR TITLE
Allow connector derived from Hive to specialize filter pushdown

### PR DESCRIPTION
**Pull request by MnO2**
## Description
Meta's internal FilterPushdown inherits the HiveFilterPushdown. Due to the
refactoring on HiveFilterPushdown, Meta's Internal build is broken. In order
to fix the build, the methods in HiveFilterPushdown needs to be moved around
and changed for visibility in order for the derived class to use the methods
in the super class.

## Impact
* No impact. It is to move the methods around.

## Test Plan
* Unit tests.

## Contributor checklist

- [x] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [x] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [x] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [x] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [x] Adequate tests were added if applicable.
- [x] CI passed.

## Release Notes

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-hive/src/test/java/com/facebook/presto/hive/AbstractTestHiveClient.java [hunk 1]
```diff
@@ -2303,14 +2304,16 @@ private static ConnectorPushdownFilterResult pushdownFilter(
...
                 session,
                 new PlanNodeIdAllocator(),
                 rowExpressionService,
                 functionResolution,
                 functionMetadataManager,
-                new HiveTransactionManager(),
-                partitionManager);
+                hiveTransactionManager,
+                partitionManager,
+                t -> HiveFilterPushdown.getConnectorMetadata(hiveTransactionManager, t));
```
Comment (tdcmeehan) [inline presto-hive/src/test/java/com/facebook/presto/hive/AbstractTestHiveClient.java:2316]:
```suggestion
                tableHandle -> HiveFilterPushdown.getConnectorMetadata(hiveTransactionManager, tableHandle));
```

**Review by tdcmeehan — CHANGES_REQUESTED** [review]
Please reword commit to something like `Allow connector derived from Hive to specialize filter pushdown`

File: presto-hive/src/test/java/com/facebook/presto/hive/AbstractTestHiveClient.java (hunk 1 above)
Comment (MnO2) [reply presto-hive/src/test/java/com/facebook/presto/hive/AbstractTestHiveClient.java:2316]: tableHandle is already defined on line 2303: https://github.com/prestodb/presto/pull/21574#diff-97caffd1bf83a7671054a8dbd37f2d7a7aebeba74d2cc02f976b4adbb2cce8aeR2303
Screenshot 2023-12-20 at 11 03 41 AM
that's why I used a different name. Do we have a naming convention for the lambda function? tableHandleForLambda? tH? or `tableHandle_`?

Comment (ajaygeorge) [reply presto-hive/src/test/java/com/facebook/presto/hive/AbstractTestHiveClient.java:2316]: just rename `ConnectorTableHandle tableHandle` var as `ConnectorTableHandle connectorTableHandle`
and keep the lambda as tableHandle

Comment (MnO2) [reply presto-hive/src/test/java/com/facebook/presto/hive/AbstractTestHiveClient.java]: @ajaygeorge @tdcmeehan I already updated the variable naming and also make the commit message more clear.

**Review by tdcmeehan — APPROVED** [review]

**Review by ajaygeorge — APPROVED** [review]
LGTM

**Pull request merged**